server/util.go

```diff
@@ -260,6 +259,8 @@ func dumpRowValuesBinary(alloc arena.Allocator, columns []*ColumnInfo, row []typ
 			nulls[bytePos] |= 1 << bitPos
 		}
 	}
+	data = make([]byte, 0, 1+nullsLen+8*len(row))
+	data = append(data, mysql.OKHeader)
 	data = append(data, nulls...)
```

Comment:
How about alloc `data` and `nulls` in slice to avoid this operation?

Reply:
good idea